### DISCUSSION TITLE
Coupling variables into FVDirichletBC

### DISCUSSION THREAD

**Discussion by ln53** [idaholab/moose#17872] [Q&A Modules: Navier-Stokes] [answered]
Hi all, I'm trying to create an FV BC which sets velocity at the boundary depending on a variable phi. However when I try to couple phi using adCoupledValue as I would in other BCs/Kernels, I get the error 'adCoupledValue was not declared in this scope'. (I've also tried reading phi into a material property and coupling it that way).
I can see the Coupleable class is not used at all by FVBoundaryCondition, and I'm also aware that in FV the variable would be defined for the element and not the boundary. But is there any way I can access the variable phi somehow for use in the BC, whether that's just using the value from the element, or somehow interpolating between the ghost elements used by FVDirichletBC?
Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

You're right that FVBCs do not inherit the coupleable interface because it does not provide the support for face values, which is what you want to be using in finite volume most of the time. 
You can add it if you want the cell value, in your boundary condition object. 

Otherwise, use var->getBoundaryFaceValue(face_info)

Best,
Guillaume

**Reply by ln53** [reply]
Thanks for the help
